server/src/frameworks/Ape/ApeProject.ts

```diff
@@ -0,0 +1,233 @@
+    } else {
+      // Project could not be initialized. Claim all files under base path to avoid them being incorrectly assigned to other projects
+      return { belongs: directoryContains(this.basePath, uri), isLocal: false };
+    }
+  }
+
+  public async resolveImportPath(file: string, importPath: string) {
+    let transformedPath = importPath;
+
+    // Apply remappings to importPath
+    for (const { from, to } of this.remappings) {
+      if (!importPath.startsWith(from)) {
+        continue;
+      }
+
+      // Extract dependency name e.g. `@openzeppelin=OpenZeppelin/4.4.2` => `OpenZeppelin`
+      const [dependencyName] = to.split(path.sep);
+
+      if (dependencyName === undefined) {
+        continue;
+      }
+
+      // Find the dependency by name by searching config's registered dependencies
+      const dependency = this.dependencies.find(
+        (d) => d.name === dependencyName
+      );
+
+      if (dependency === undefined) {
+        continue;
+      }
+
+      // Dependencies' subfolders are different if they are specified by version or branch
+      const dependencyBasePath = this._resolveDependencyBasePath(dependency);
+
+      transformedPath = importPath.replace(
+        from.endsWith("/") ? from : `${from}/`,
+        `${dependencyBasePath}/`
+      );
+    }
+
+    // Try to resolve the import recursively, start from source directory up to project root
+    let testBaseDirectory = path.dirname(file);
+    let resolvedPath: string | undefined;
+
+    while (directoryContains(this.basePath, testBaseDirectory)) {
+      const testResolvedPath = path.resolve(testBaseDirectory, transformedPath);
+
+      if (existsSync(testResolvedPath)) {
+        resolvedPath = testResolvedPath;
+        break;
+      }
+
+      testBaseDirectory = path.dirname(testBaseDirectory);
+    }
+
+    return resolvedPath !== undefined ? toUnixStyle(resolvedPath) : undefined;
+  }
+
+  public async buildCompilation(
+    sourceUri: string,
+    openDocuments: OpenDocuments
+  ): Promise<CompilationDetails> {
+    // Ensure project is initialized
+    if (this.initializeError !== undefined) {
+      const error: InitializationFailedError = {
+        _isInitializationFailedError: true,
+        error: this.initializeError,
+      };
+
+      throw error;
+    }
+
+    const basicCompilation = await buildBasicCompilation(
+      this,
+      sourceUri,
+      openDocuments,
+      this.configSolcVersion
+    );
+
+    return basicCompilation;
+  }
+
+  public async onWatchedFilesChanges({
+    changes,
+  }: DidChangeWatchedFilesParams): Promise<void> {
+    for (const change of changes) {
+      if (change.uri === this.configPath) {
+        this.serverState.logger.info(
+          `Reinitializing Ape project: ${this.id()}`
+        );
+        await this.initialize();
+      }
+    }
+    return;
+  }
+
+  public getImportCompletions(
+    _position: Position,
+    _currentImport: string
+  ): CompletionItem[] {
+    return [];
```

Comment:
What would be involved in getting import completions.

Reply:
Probably iterating over registered dependencies and scanning them for contracts. I left it for a next iteration like we did with foundry's adapter.